commands2/command.py

```diff
@@ -0,0 +1,61 @@
+from abc import abstractmethod
+from typing import Generator, Callable, Set, Optional
+
+from .requirement import Requirement, InterruptBehavior
+
+
+class Command:
```

Comment:
Depends on when we want errors to be caught I guess. If we want errors when instantiating the class then an ABC is appropriate. Otherwise if we want duck typing then a Protocol will provide that.

If there's only one method, and no custom constructor, probably no huge advantage to an ABC?